### PR TITLE
Prepare for release 4.4.0-beta1.

### PR DESCRIPTION
**Pull request by f2prateek**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,25 @@
...
+ ```
+
+ * [New](https://github.com/segmentio/analytics-android/pull/582): Send a custom user agent on HTTP requests originating the library. The new user-agent will be formatted as `analytics-android/{version}`.
+
+ * [Fix](https://github.com/segmentio/analytics-android/pull/578): Retry network requests when server responds with a HTTP 429 response code.
+
+ * [Fix](https://github.com/segmentio/analytics-android/pull/575): Update maximum message limit to be 32kb to match our API limits.
+
+ * [Fix](https://github.com/segmentio/analytics-android/pull/573): Fix race condition that could cause events to pick up user IDs after the event was originally recorded.
+
```
Comment (nhi-nguyen) [inline CHANGELOG.md]: Should this https://github.com/segmentio/analytics-android/issues/481 be included?

Comment (f2prateek) [reply CHANGELOG.md]: Yep - it's the first one I called out (as it's somewhat of a breaking change).

Comment (f2prateek) [reply CHANGELOG.md]: Oh I labelled it incorrectly though, fixing now.

Comment (f2prateek) [reply CHANGELOG.md]: Actually sorry, #574 is the right PR, #481 is the original issue but not the PR.

Comment (nhi-nguyen) [reply CHANGELOG.md]: LGTM. May be useful to add a comment in the original issues about this version in case anyone is watching them for a fix.

Comment (f2prateek) [reply CHANGELOG.md]: Added a note!

**Pull request merged**